misc/identity-keys/src/identity-keys.ts

```diff
@@ -0,0 +1,162 @@
+import * as ed25519 from "@noble/ed25519";
+import { Cacao } from "@walletconnect/cacao";
+import { Store } from "@walletconnect/core";
+import {
+  JwtPayload,
+  composeDidPkh,
+  encodeEd25519Key,
+  generateJWT,
+  jwtExp,
+} from "@walletconnect/did-jwt";
+import { ICore, IStore } from "@walletconnect/types";
+import { formatMessage, generateRandomBytes32 } from "@walletconnect/utils";
+import axios from "axios";
+import {
+  IIdentityKeys,
+  IdentityKeychain,
+  RegisterIdentityParams,
+  ResolveIdentityParams,
+  UnregisterIdentityParams,
+} from "./types";
+
+const DEFAULT_KEYSERVER_URL = "https://keys.walletconnect.com/";
+const IDENTITY_KEYS_STORAGE_PREFIX = "wc@2:identityKeys";
+
+export class IdentityKeys implements IIdentityKeys {
+  private keyserverUrl: string;
+  private identityKeys: IStore<IdentityKeychain["accountId"], IdentityKeychain>;
+
+  constructor(private core: ICore, keyServerUrl?: string) {
+    this.keyserverUrl = keyServerUrl ?? DEFAULT_KEYSERVER_URL;
+    this.identityKeys = new Store(
```

Comment:
Having a separate st

Reply:
While you were OOO, Cali and I realised that what we discussed initially created more problems than it solved. Something we didn’t properly account for in our original discussion was that our `Store` interface itself consumes the core (all existing stores live at client level not core level).

We decided that it’s simpler and keeps modularity to treat this class the same way and let it manage a store, removing any additional burden and complexity on the core controller.

You can see the awkward situation Cali ran into here:
https://github.com/WalletConnect/walletconnect-monorepo/pull/2190

Reply:
Details like that do come up in implementation, thank you for clearing it up